flink-table-store-core/src/main/java/org/apache/flink/table/store/file/format/FileFormatImpl.java

```diff
@@ -50,7 +54,12 @@ protected BulkDecodingFormat<RowData> getDecodingFormat() {
     }
 
     @Override
-    protected EncodingFormat<BulkWriter.Factory<RowData>> getEncodingFormat() {
-        return writerFactory.createEncodingFormat(null, formatOptions); // context is useless
+    public FormatWriter.Factory<RowData> createWriterFactory(RowType writeSchema) {
+        BulkWriter.Factory<RowData> bulkWriter =
+                writerFactory
+                        .createEncodingFormat(null, formatOptions)
+                        .createRuntimeEncoder(SINK_CONTEXT, fromLogicalToDataType(writeSchema));
+
+        return new RowFormatWriter.RowFormatWriterFactory(bulkWriter, writeSchema, null);
```

Comment:
The default `FileFormatImpl` don't have any general `FileStatsExtractor`, so we just use the `null` by default. Finally, it will just use the `FieldStatsCollector` to extract the column max-min.